lib/Wikia/src/Service/User/Preferences/PreferenceModule.php

```diff
@@ -27,8 +30,22 @@ public function configure( InjectorBuilder $builder ) {
 				global $wgHiddenPrefs;
 				return $wgHiddenPrefs;
 			} )
-			->bind( PreferenceServiceImpl::DEFAULT_PREFERENCES )->to( function() {
-				return User::getDefaultPreferences();
+			->bind( PreferenceServiceImpl::DEFAULT_PREFERENCES )->to( function( ContainerInterface $c ) {
+				/** @var PreferenceScopeService $scopeService */
+				$scopeService = $c->get( PreferenceScopeService::class );
+				$defaultOptions = User::getDefaultOptions();
+				$defaultPreferences = new UserPreferences();
+
+				foreach ( $defaultOptions as $name => $val ) {
+					if ( $scopeService->isGlobalPreference( $name ) ) {
+						$defaultPreferences->setGlobalPreference( $name, $val );
+					} elseif ( $scopeService->isLocalPreference( $name ) ) {
+						list( $prefName, $wikiId ) = $scopeService->splitLocalPreference( $name );
+						$defaultPreferences->setLocalPreference( $prefName, $wikiId, $val );
+					}
+				}
+
+				return $defaultPreferences;
```

Comment:
I'd say it is, since nothing else in the codebase needs this `UserPreferences` object. It's only needed to instantiate the `PreferenceService`. I'd say this is pretty much equivalent to something like this using Guice:

```java
class MyClass {
  @Inject
  public MyClass(OtherClass otherClass) {
    ..
  }
}
...
class MyModule extends AbstractModule {
  @Provides
  public OtherClass providesOtherClass(MyConfiguration config) {
    // build OtherClass from stuff inside MyConfiguration
  }
}
```